### PR TITLE
Fix objectNames when get JmxTableHandle

### PR DESCRIPTION
**Pull request by kyo-tom**
Description
Additional context and related issues
Release notes
( ) This is not user-visible or docs only and no release notes are required.
( ) Release notes are required, please propose a release note for me.
(x) Release notes are required, with the following suggested text:

Fix https://github.com/trinodb/trino/issues/16037

### CODE REVIEW

**Comment by Praveen2112** [comment]
Thanks for working on this. Can we add some tests for this PR ?

**Comment by kyo-tom** [comment]
> Thanks for working on this. Can we add some tests for this PR ?

Of course, we need to check if our `objectNames` in `tableHandle` exist in the table list of `JmxMetadata`.

**Comment by kyo-tom** [comment]
PTAL @Praveen2112

File: plugin/trino-jmx/src/main/java/io/trino/plugin/jmx/JmxMetadata.java [hunk 1]
```diff
@@ -152,7 +152,7 @@ private JmxTableHandle getJmxTableHandle(SchemaTableName tableName)
                     .sorted(comparing(JmxColumnHandle::getColumnName))
                     .collect(toImmutableList());
 
-            return new JmxTableHandle(tableName, objectNames.stream().map(ObjectName::toString).collect(toImmutableList()), columns, true, TupleDomain.all());
+            return new JmxTableHandle(tableName, objectNames.stream().map(ObjectName::getCanonicalName).collect(toImmutableList()), columns, true, TupleDomain.all());
```
Comment (Praveen2112) [inline plugin/trino-jmx/src/main/java/io/trino/plugin/jmx/JmxMetadata.java:155]: I'm not sure if we need to use `getCanonicalName` name here as it represents the name of the object and it is a part of the data.

Comment (kyo-tom) [reply plugin/trino-jmx/src/main/java/io/trino/plugin/jmx/JmxMetadata.java:155]: `getCanonicalName` will returns the canonical form of the name; that is, a string representation where the properties are sorted in lexical order. 
`toString ` will return the name and the key properties in the returned string will be in the same order as in the argument to the constructor.
image

Comment (kyo-tom) [reply plugin/trino-jmx/src/main/java/io/trino/plugin/jmx/JmxMetadata.java:155]: I don't think it matters which table name is used, what matters is that we need to be consistent with table name in `JmxHistoricalData`

Comment (kyo-tom) [reply plugin/trino-jmx/src/main/java/io/trino/plugin/jmx/JmxMetadata.java:155]: cc @Praveen2112

**Comment by mosabua** [comment]
:wave: @kyo-tom @Praveen2112  - this PR has become inactive. We hope you are still interested in working on it. Please let us know, and we can try to get reviewers to help with that.

We're working on closing out old and inactive PRs, so if you're too busy or this has too many merge conflicts to be worth picking back up, we'll be making another pass to close it out in a few weeks.

**Pull request closed without merging**